### PR TITLE
Add docs

### PR DESCRIPTION
**Pull request by mononoken**
# 🔗 Issue
#806

# ✍️ Description
I think this is a good starting point to adding some architectural docs to the project. These notes are from the meeting this weekend. I don't think we need to be adding a ton of notes here, this will be mostly for big architectural decisions.

### CODE REVIEW

File: doc/architecture/002-person-model.md
Comment (jmilljr24) [inline doc/architecture/002-person-model.md:14]: For clarity, I believe `OrganizationProfile` columns were going to be added back to `Organization`.

File: doc/architecture/003-checklists.md
Comment (jmilljr24) [inline doc/architecture/003-checklists.md:28]:
```suggestion
`Assignment` will link the `TemplateItem` with a `Person` or `Pet`. The `Assignment` will cache the `TemplateItem` instead of soft linking to it. That way, if the `TemplateItem` is modified, the `Assignment` is unaffected.
```

File: doc/architecture/003-checklists.md
Comment (jmilljr24) [inline doc/architecture/003-checklists.md:12]:
```suggestion
- Ken: Currently, we have `Task` as something that is open for any staff member to complete. A `Task` is created by a staff member for a specific pet. Any other staff member can look at the pet and mark the task as completed. In the new structure, a `Checklist::Assignment` is associated with a `Person` or `Pet`. This is a fundamental difference, and I am not sure which is preferred.
```

**Review by jmilljr24 — APPROVED** [review]
Thanks for writing this up!!

**Comment by mononoken** [comment]
Thanks for looking over these, @jmilljr24 ! I will wait until after the meeting to merge these, so that we can discuss them first.

**Comment by mononoken** [comment]
@kasugaijin and @jmilljr24 , I finally got around to updating this. If anything looks wrong based on the recent meetings or if you want anything added, let me know.

File: doc/architecture/003-checklists.md
Comment (kasugaijin) [inline doc/architecture/003-checklists.md:13]: Could add here that Task currently serves as the 'Assignment'.

File: doc/architecture/001-document-decisions.md
Comment (kasugaijin) [inline doc/architecture/001-document-decisions.md:5]: Should we keep this document tethered to the current code conditions only? I.e., describes what the code does right now, without historical or future contexts. I think that is a good compromise. It keeps things shorter (more maintainable) and doesn't dilute anything for the new developer (e.g., addingin in additional contexts).

**Review by kasugaijin — APPROVED** [review]
Looks good, thanks @mononoken 
No blocking suggestions for changes

**Pull request merged**
